File: src/Modules/Owners/AddModRole.cs
using Discord;
using Discord.Commands;
using System.Threading.Tasks;
using System.Linq;
using MongoDB.Driver;

namespace DEA.Modules.Owners
{
    public partial class Owners
    {
        [Command("AddModRole")]
        [Summary("Adds a moderator role.")]
        public async Task AddModRole(IRole modRole, int permissionLevel = 1)
        {
            if (permissionLevel < 1 || permissionLevel > 3)
            {
                ReplyError("Permission levels:\nModeration: 1\nAdministration: 2\nServer Owner: 3");
            }

            if (Context.DbGuild.ModRoles.ElementCount == 0)
            {
                await _guildRepo.ModifyAsync(Context.DbGuild, x => x.ModRoles.Add(modRole.Id.ToString(), permissionLevel));
            }
            else
            {
                if (Context.DbGuild.ModRoles.Any(x => x.Name == modRole.Id.ToString()))
                {
                    ReplyError("You have already set this mod role.");
                }

                await _guildRepo.ModifyAsync(Context.DbGuild, x => x.ModRoles.Add(modRole.Id.ToString(), permissionLevel));
            }

            await ReplyAsync($"You have successfully added {modRole.Mention} as a moderation role with a permission level of {permissionLevel}.");
        }
    }
}

File: src/commands/owners/AddModRole.js
const db = require('../../database');
const patron = require('patron.js');
const util = require('../../utility');

class AddModRole extends patron.Command {
  constructor() {
    super({
      name: 'addmodrole',
      aliases: ['addmod', 'setmod'],
      group: 'owners',
      description: 'Add a mod role.',
      args: [
        new patron.Argument({
          name: 'role',
          key: 'role',
          type: 'role',
          example: 'Moderator'
        }),
        new patron.Argument({
          name: 'permissonLevel',
          key: 'permissionLevel',
          type: 'float',
          example: '2'
        })
      ]
    });
  }

  async run(msg, args){
    if (args.permissionLevel < 1 || args.permissionLevel > 3) {
      return util.Messenger.replyError(msg.channel, msg.author, 'Permission levels:\nModerator: 1\nAdministrator: 2\nOwner: 3');
    } else if (msg.dbGuild.roles.mod.some((role) =>  role.id === args.role.id)) {
      return util.Messenger.replyError(msg.channel, msg.author, 'This moderation role has already been set.');
    }

    await db.guildRepo.upsertGuild(msg.guild.id, new db.updates.Push('roles.mod', { id: args.role.id, permissionLevel: args.permissionLevel }));

    return util.Messenger.reply(msg.channel, msg.author, 'You have successfully added the mod role ' + args.role + ' with a permission level of ' + args.permissionLevel + '.');
  }
}

module.exports = new AddModRole();
